conda/cli/main.py

```diff
@@ -91,6 +91,7 @@ def main(*args):
     args = tuple(ensure_text_type(s) for s in args)
 
     if len(args) > 1:
+        from ..exceptions import HelpError
```

Comment:
I don't think this PR is generally the right direction, and this line illustrates why.  The activate stuff has to be blazing fast, and imports end up costing us a lot.  The `conda/activate.py` file is already set up to support informational help strings appropriately.  We just need to fill out the content of those strings.  For example,

```
$ conda activate -h
help requested for activate
```

So that just needs content.  And for the `activate` and `deactivate` commands missing from `conda --help`, we can do something like we do with the "other commands" message overriding the `print_help()` function.  Just add some extra information in there to output about those two commands.

Reply:
I should say that I'm also hostile to argparse in general, because of how much work it does to get all set up on every invocation of the program.  It's something we can live with (and just have to live with) for conda commands that are expected to take a bit longer, but for things like activate that potentially affect the startup time of shells (because `conda activate base` will be part of most people's shell setup), we *have* to be fast.

Reply:
I was expecting you to say something along those lines. I guess my main point in trying to do it this way was to bring content together. There are definitely bits and pieces of conda that feels (or in the case of the old source scripts, felt) fragmented.

What if we were even more reluctant to import `HelpError`, hence it only gets imported inside the except clause (and then use an `isinstance` check)? This way we wont expect any slowdown for successful cases but a slight slowdown for error cases or cases requesting more info.

Adding the help info for activate and deactivate in a unique, separate manner from everything else feels like we're reinventing the wheel and makes the implementation feel fragmented again...

Reply:
> I was expecting you to say something along those lines.

Well that kind of make me smile 😁

Ok, how about this... We already have help information in the `conda/exceptions.py` module, as you've probably noticed.  `CommandNotFoundError` for example.  I guess the whole exceptions module is really about user messaging, which makes since the help information could be there too.

So only line 106/109 below, 

    except Exception as e:

change the content after that to do type checking on the exception instance `e`.

```python
from ..exceptions import HelpError
if isinstance(e. HelpError):
```

Actually, scratch that...  All we need to do is a local import of `HelpError` in `conda/activate.py`.  And raise that.  Everything else will take care of itself then.  And we can still have the help information in `exceptions.py`.

Reply:
So you're suggesting that instead of the help information being part of the argparse to make an exception that contains/prints the help info?

Do we want argparse to use this same strategy for printing the help or would there be two separate help implementations - a classic argparse print_help and an exception help?